### PR TITLE
 [MRESOLVER-85] Cast ByteBuffer to Byte for JDK8 support 

### PR DESCRIPTION
**Pull request by slachiewicz**

### CODE REVIEW

File: maven-resolver-connector-basic/src/main/java/org/eclipse/aether/connector/basic/ChecksumCalculator.java [hunk 1]
```diff
@@ -199,9 +201,10 @@ public void update( ByteBuffer data )
     {
         for ( Checksum checksum : checksums )
         {
-            data.mark();
+            // Cast ByteBuffer to Byte for JDK8 support
+            ( ( Buffer ) data ).mark();
```
Comment (michael-o) [inline maven-resolver-connector-basic/src/main/java/org/eclipse/aether/connector/basic/ChecksumCalculator.java]: The spaces around `Buffer` aren't necessary.

File: .travis.yml [hunk 2]
```diff
@@ -19,7 +19,7 @@ before_script: true
 install: true
 
 script:
-  - mvn verify
+  - mvn -V -P+run-its clean verify
```
Comment (michael-o) [inline .travis.yml]: `+run-its`?

File: maven-resolver-connector-basic/src/main/java/org/eclipse/aether/connector/basic/ChecksumCalculator.java (hunk 1 above)
Comment (slachiewicz) [reply maven-resolver-connector-basic/src/main/java/org/eclipse/aether/connector/basic/ChecksumCalculator.java]: ok, updated.

**Comment by slachiewicz** [comment]
I don't have rights to push to maven-resolver branch to run in ASF Jenkins, but all test passed locally with Java 12. 
Also, tests running on Linux with Travis CI - passed without issues for OpenJDK 7, 8, 11, 12.

https://travis-ci.org/slachiewicz/maven-resolver/builds/519294192

**Comment by michael-o** [comment]
This is wrong, you should be able to do so. Can you raise this on the dev list?

**Comment by slachiewicz** [comment]
run-its - i used the value from command line in latest Jenkins build log.
>
>[Pipeline] bat
>Commit message: "[MRESOLVER-7] Download dependency POMs in parallel"
>
>f:\jenkins\jenkins-slave\workspace\maven-box_maven-resolver_master\m>mvn -P+run-its -Dmaven.test.failure.ignore=true -Dfindbugs.failOnError=false -Dfindbugs.skip=true clean verify 
>----- withMaven Wrapper script -----
>
>f:\jenkins\jenkins-slave\workspace\maven-box_maven-resolver_master@2\m>mvn -P+run-its -Dmaven.test.failure.ignore=true -Dfindbugs.failOnError=false -Dfindbugs.skip=true clean verify 
>----- withMaven Wrapper script -----
>Cleaning workspace
>git clean -fdx # timeout=10
>Apache Maven 3.5.2 (138edd61fd100ec658bfa2d307c43b76940a5d7d; 2017-10-18T07:58:13Z)
>Maven home: F:\jenkins\tools\maven\apache-maven-3.5.2\bin\..

**Comment by michael-o** [comment]
That's nonsense, look into the POM. There is no such profile.

**Comment by slachiewicz** [comment]
> That's nonsense, look into the POM. There is no such profile.

But you asked to run its :)

here is profile..
https://github.com/apache/maven-resolver/blob/master/maven-resolver-demos/maven-resolver-demo-maven-plugin/pom.xml#L87

**Comment by michael-o** [comment]
Indeed, but the `+` is still wrong.

**Comment by slachiewicz** [comment]
> Indeed, but the `+` is still wrong.

so all our builds should be wrong

https://gitbox.apache.org/repos/asf?p=maven-jenkins-lib.git;a=blob;f=vars/asfMavenTlpStdBuild.groovy;h=8bf4492dd33d5a226ed0f05b549d6f48f20767ad;hb=HEAD#l55

**Comment by jira-importer** [comment]
Resolve #826

**Comment by jira-importer** [comment]
Resolve #826

**Pull request closed without merging**